solace-spring-cloud-stream-binder/solace-spring-cloud-stream-binder-core/src/test/java/com/solace/spring/cloud/stream/binder/util/XMLMessageMapperTest.java

```diff
@@ -1071,7 +1076,14 @@ public void testMapXMLMessageToSpringMessage_ReadLocalSolaceProperties() throws
 		Mockito.verify(xmlMessageMapper).map(xmlMessage, acknowledgmentCallback, false);
 
 		for (Map.Entry<String, ? extends HeaderMeta<?>> header : readableLocalHeaders) {
-			fail(String.format("no test for header %s", header.getKey()));
+			Object actualValue = springMessage.getHeaders().get(header.getKey());
+			switch (header.getKey()) {
+				case SolaceBinderHeaders.NULL_PAYLOAD:
```

Comment:
Can't add a comment there for some reason, but on line `1060`, can you uncomment the `assertNotEquals("Test header set was empty", 0, readableLocalHeaders.size());`? Now that we have actual local-scoped readable headers, we can actually start running that assertion. 😆

Reply:
You were forward thinking when you wrote that ;)